website/src/pages/fonts/[font].tsx

```diff
@@ -58,19 +75,54 @@ export default function FontPage({ metadata, defPreviewText }: FontPageProps) {
 
 export const getStaticProps: GetStaticProps = async ({ params }) => {
   try {
-    const metadata: MetadataProps = await fetcher(
+    const metadata: MetadataProps = await fetchJson(
       fontsourceDownload.data(`${params.font}`).metadata
     );
 
+    if (!metadata) return { notFound: true };
+
     // Generates preview texts in matching languages
     const defPreviewText = selectDefPreviewText(
       metadata.fontId,
       metadata.defSubset
     );
 
-    return metadata
-      ? { props: { metadata, defPreviewText }, revalidate: 7200 }
-      : { notFound: true };
+    const fontCssNames: [number, string[]][] = [];
+    // Fetch css data
+    const fontCssData: string[][] = await Promise.all(
+      metadata.weights.map((weight) => {
+        fontCssNames.push([weight, []]);
+        return Promise.all(
+          metadata.styles.map((style) => {
+            fontCssNames[fontCssNames.length - 1][1].push(style);
+
+            return fetchText(
+              fontsourceDownload.cssDownload(metadata.fontId, weight, style)
+            );
+          })
+        );
+      })
+    );
+
+    const fontCss: FontPreviewCss = {};
+
+    // Correctly map css data to its corispoding type (weight/style), while minimizing the css.
+    fontCssNames.forEach((weight, weightIndex) => {
+      fontCss[weight[0]] = {};
+
+      weight[1].forEach((style, styleIndex) => {
+        fontCss[weight[0]][style] = new CleanCSS()
+          // minify css useing the "clean-css" package
+          .minify(fontCssData[weightIndex][styleIndex])
```

Comment:
Rather than adding the overhead of an entirely new dependency to minify the CSS, why not just strip the whitespace? CleanCSS isn't actually doing any other transformations to the CSS other than that.